jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryAuthenticator.java

```diff
@@ -103,6 +103,15 @@ public static Initializer initializer(String serverUrl, String repository) {
     return new Initializer(serverUrl, repository);
   }
 
+  @Nullable
+  private static Integer getHttpTimeout() {
```

Comment:
Is using `null` a common way to do this? If not, I was thinking maybe use negative to indicate absence?

Reply:
I think it's not uncommon to use `null` to indicate a field is unset or uninitialized, or something is absent. Using a negative value works too. However, it won't be distinguishable from the user wrongly giving a negative value. That will matter if we error out on a negative timeout value as you said.

Reply:
Okay sounds good :+1: let's add a javadoc explicitly mentioning that `null` means the caller should use some default.

Reply:
Ah, you weren't talking about a field here. I see what you meant. I'll think about the API-aspect of this method.